dask_expr/_util.py

```diff
@@ -19,6 +20,41 @@
 DASK_GT_20231000 = DASK_VERSION > Version("2023.10.0")
 
 
+def _maybe_shift_divisions(df, periods, freq):
+    """Maybe shift divisions by periods of size freq
+
+    Used to shift the divisions for the `shift` method. If freq isn't a fixed
+    size (not anchored or relative), then the divisions are shifted
+    appropriately. Otherwise the divisions are cleared.
+
+    Parameters
+    ----------
+    df : dd.DataFrame, dd.Series, or dd.Index
+    periods : int
+        The number of periods to shift.
+    freq : DateOffset, timedelta, or time rule string
+        The frequency to shift by.
+    """
+    if isinstance(freq, str):
+        freq = pd.tseries.frequencies.to_offset(freq)
+
+    is_offset = isinstance(freq, pd.DateOffset)
+    if is_offset:
+        if freq.is_anchored() or not hasattr(freq, "delta"):
+            # Can't infer divisions on relative or anchored offsets, as
+            # divisions may now split identical index value.
+            # (e.g. index_partitions = [[1, 2, 3], [3, 4, 5]])
+            # return df.clear_divisions()
+            return None
```

Comment:
Should this result In all None divisions?

Reply:
Depends on the caller, for _divisions it'll return from here or use the existing .frame.divisions. If it's dispatching to MapPartitions then it'll set `clear_divisions` if this returns None.

Reply:
Yikes ok

Reply:
Updated the `_divisions` method when this returns none.
https://github.com/dask-contrib/dask-expr/pull/473/commits/676330ef2cb6d3313873f797625f4bc5d40eae48